PythonAPI/examples/autoware_demo.py

```diff
@@ -306,67 +423,39 @@ def main():
     argparser.add_argument(
         '--follow', action='store_true',
         help='Follow Ego vehicle')
+    argparser.add_argument(
+        '--resync_mode', action='store_true',
+        help='Resynchronize to the current time when lag exceeds the acceptable threshold')
+    argparser.add_argument(
+    	'--load_map', nargs='?', const='Town10HD_Opt',
+    	help='Load the provided map')
     args = argparser.parse_args()
 
+    # Get Client info
     client = carla.Client(args.host, args.port)
     client.set_timeout(60.0)
     world = client.get_world()
 
-    # Autoware will be publishing TF information based on the URDF files
-    # of the vehicle and sensor kit. Disable TF publishing in CARLA
-    # to avoid conflicts.
-    world.set_publish_tf(False)
+    # Determine which map to load
+    map_name = args.load_map if args.load_map is not None else 'Town10HD_Opt'
 
+    # Apply Settings
+    apply_world_settings(client, world, map_name)
+    log_info("Simulation time scale is %f" % args.time_scale)
+
+    # Spawn Ego
     spawn_point = random.choice(world.get_map().get_spawn_points())
     ego = spawn_ego_with_sensors(world, spawn_point)
-    move_spectator(world, ego)
-
-    info('Ego spawned!')
-    time.sleep(0.05)  # Without this sometimes spectator would not move
 
-    # Set synchronous mode
-    settings = world.get_settings()
-    settings.synchronous_mode = True
-    settings.fixed_delta_seconds = 1.0 / DESIRED_SIM_RATE
-    world.apply_settings(settings)
-
-    info("Simulation time scale is %f" % args.time_scale)
-
-    warning('Kill this script before stopping simulation!')
-
-    frame_count = 0
-
-    start_time = time.time()
-    last_behind_error_time = start_time
-    # Tick world to follow desired time_scale
-    try:
-        while True:
-            current_time = time.time()
-            real_time_passed = current_time - start_time
-            sim_time_passed = real_time_passed * args.time_scale
-            desired_frame_count = math.floor(sim_time_passed * DESIRED_SIM_RATE)
+    world.tick()  # tick to process the changes (settings, ego + sensors spawn)
+    move_spectator(world, ego)
 
-            if desired_frame_count <= frame_count:
-                continue
+    log_info('Ego spawned!')
+    log_warning('Kill this script before stopping simulation!')
 
-            frame_behind_count = max(desired_frame_count - frame_count, 0)
-            if (frame_behind_count > 10) and \
-                (current_time > last_behind_error_time + 2.0):  # Error every 2 seconds
-                    last_behind_error_time = current_time
-                    error(f'Simulation cannot keep up with the desired time scale!!! ({frame_behind_count} frames behind)')
+    # Run simulation loop
+    run_simulation_loop(target_time_scale=args.time_scale, follow_ego=True, world=world, ego=ego, should_resync=args.resync_mode)
```

Comment:
Suggestion: split a long argument list so that each argument is in a separate line.
This is consistent with the rest of the source code within that file.

```suggestion
    run_simulation_loop(
        target_time_scale=args.time_scale,
        follow_ego=True,
        world=world,
        ego=ego,
        should_resync=args.resync_mode)
```

Reply:
Resolved in #37